NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/Implementation/RoleNameContainer.cs

```diff
@@ -0,0 +1,66 @@
+﻿namespace Microsoft.ApplicationInsights.AspNetCore.Implementation
+{
+    using System;
+    using System.Threading;
+
+    using Microsoft.AspNetCore.Http;
+
+    internal static class RoleNameContainer
+    {
+        private const string WebAppHostNameHeaderName = "WAS-DEFAULT-HOSTNAME";
+        private const string WebAppHostNameEnvironmentVariable = "WEBSITE_HOSTNAME";
+
+        private static string roleName = string.Empty;
+
+        public static string RoleName
+        {
+            get => roleName;
+
+            set
+            {
+                if (value != roleName)
```

Comment:
why not using Interlocked.CompareExchange for comparison and exchange?

Reply:
Simple answer is, i didn't know the correct way to use CompareExchange.
Now that I have better test coverage, i'll take another look at this. :)

Reply:
If you know something I don't, please share. :)

From what I'm reading, CompareExchange will always replace the `ref variable` with an incoming value.
In my use case, I only need to replace the `ref variable` when the incoming value does not match.
Specifically, when a VIP SWAP occurs, we need to update the role name.
My understanding is, replacing this with a CompareExchange will always replace the `ref variable` with the incoming value regardless of if it's new.

I believe CompareExchange doesn't help me in this use case.
But please correct me if i'm wrong.